src/Microsoft.ML.Data/DataView/TypedCursor.cs

```diff
@@ -231,6 +243,9 @@ public static TypedCursorable<TRow> Create(IHostEnvironment env, IDataView data,
             env.AssertValue(data);
             env.AssertValueOrNull(schemaDefinition);
 
+            if (schemaDefinition == null)
+                CheckTypeWithOnnx(typeof(TRow), data);
```

Comment:
Do we want to call the function even if the schemaDefinition is not null? Can you check where else this function is called from with a non-null schemaDefinition?

Reply:
yeah there could be existing schemaDefinition so I added that part in case something happened if schemaDefinition is not null